datafusion/expr/src/udaf.rs

```diff
@@ -552,6 +570,14 @@ pub trait AggregateUDFImpl: Debug + Send + Sync {
     fn is_descending(&self) -> Option<bool> {
         None
     }
+
+    /// Returns default value of the function given the input is all `null`.
+    ///
+    /// Most of the aggregate function return Null if input is Null,
+    /// while `count` returns 0 if input is Null
+    fn default_value(&self, data_type: &DataType) -> Result<ScalarValue> {
+        ScalarValue::try_from(data_type)
+    }
 }
```

Comment:
If an `AggregateUDFImpl ` overrides `is_nullable()` to return false but does not set the `default_value()`, it seems that it would indicate it is not nullable, yet its `default_value()` would return `null`. Is there a way to prevent this behavior? 🤔

Reply:
In this case, it should at least failed when creating record batch. Since the schema indicates non-null but got null value. 

If you are saying prevent `panic`, it is a good question, I don't have answer

Reply:
Maybe we can improve the docuemention?